controllers/validator.go

```diff
@@ -12,9 +12,34 @@ import (
 	"github.com/go-logr/logr"
 	oadpv1alpha1 "github.com/openshift/oadp-operator/api/v1alpha1"
 	"github.com/openshift/oadp-operator/pkg/credentials"
+
+	appsV1 "k8s.io/api/apps/v1"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
 func (r *DPAReconciler) ValidateDataProtectionCR(log logr.Logger) (bool, error) {
+	// When updating from OADP version 1.1.x to 1.2.x, delete
+	// daemonset.apps/restic (it was renamed to daemonset.apps/node-agent)
+	// resticrepositories.velero.io CRD (it was renamed to backuprepositories.velero.io)
+	if err := r.Delete(r.Context, &appsV1.DaemonSet{ObjectMeta: metaV1.ObjectMeta{
+		Name: "restic", Namespace: r.NamespacedName.Namespace,
+	}}); err != nil && !apiErrors.IsNotFound(err) {
+		return false, err
+	}
+	resticRepositoriesResource := &unstructured.Unstructured{}
+	resticRepositoriesResource.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "apiextensions.k8s.io",
+		Version: "v1",
+		Kind:    "CustomResourceDefinition",
+	})
+	resticRepositoriesResource.SetName("resticrepositories.velero.io")
```

Comment:
Should we not make sure that there are no CRs present in the cluster before deleting the CRD ? Or Do we already do that ?

Reply:
Did not think about this :grimacing: 
I hope upgrade kills them for us

But will try to test tomorrow if I can see how it works (install stable 1.1 -> create DPA with restic -> upgrade to stable 1.2)

Reply:
yeah let me know about this

Reply:
Tested @shubham-pampattiwar 
Upgrade does not kill CR instances of the `resticrepositories.velero.io` CRD

So we will probably need to kill them before killing CRD (but this is not in this PR anymore)

Reply:
ok, thank you! @mateusoliveira43 ++